api/main.py

```diff
@@ -69,16 +69,19 @@ def read_last_emails(email: str, count = None) -> list:
     finally:
         g.close()
 
-def write_to_vdb(mapped: list) -> None:
+def write_to_vdb(mapped: list[dict]) -> None:
+    print("Writing to VDB..." + str(len(mapped)))
     try:
         w = weaviate.Weaviate("http://127.0.0.1:8080")
         for j,email in enumerate(mapped):
             print("=> Considering email " + str(j) + " of " + str(len(mapped)) + "...")
             try:
-                w.upsert(email)
+                if email['body'] == None or email['body'] == '':
+                    email['body'] = email['subject']
+                w.upsertChunkedText(email, weaviate.WeaviateSchemas.EMAIL_TEXT, weaviate.WeaviateSchemas.EMAIL, 'body')
```

Comment:
There are now multiple models with a cross reference relationship so we can retrieve and filter on things like date and sender. Only the email text model gets chunked and vector indexed